File: app/src/main/java/com/perflyst/twire/activities/ThemeActivity.java
package com.perflyst.twire.activities;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.perflyst.twire.activities.main.MainActivity;
import com.perflyst.twire.model.Theme;
import com.perflyst.twire.service.Settings;

/**
 * Created by Sebastian Rask on 30-04-2016.
 */
public class ThemeActivity extends AppCompatActivity {
    private Theme theme;

    @Override
    protected void onCreate(Bundle savedInstance) {
        loadTheme();
        super.onCreate(savedInstance);
    }

    @Override
    public void onResume() {
        super.onResume();

        Theme currentTheme = Settings.getTheme();
        if (!currentTheme.equals(theme)) {
            recreate();
        }
    }

    private void loadTheme() {
        this.theme = Settings.getTheme();
        setTheme(theme.style);
    }

    @Override
    public void recreate() {
        if (this instanceof MainActivity) {
            ((MainActivity) this).getRecyclerView().scrollToPosition(0);
        }
        super.recreate();
    }
}

File: app/src/main/java/com/perflyst/twire/activities/ThemeActivity.kt
package com.perflyst.twire.activities

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.perflyst.twire.activities.main.MainActivity
import com.perflyst.twire.model.Theme
import com.perflyst.twire.service.Settings

/**
 * Created by Sebastian Rask on 30-04-2016.
 */
open class ThemeActivity : AppCompatActivity() {
    private var theme: Theme? = null

    override fun onCreate(savedInstance: Bundle?) {
        loadTheme()
        super.onCreate(savedInstance)
    }

    public override fun onResume() {
        super.onResume()

        val currentTheme = Settings.theme
        if (currentTheme != theme) {
            recreate()
        }
    }

    private fun loadTheme() {
        this.theme = Settings.theme
        setTheme(theme!!.style)
    }

    override fun recreate() {
        if (this is MainActivity<*>) {
            this.recyclerView.scrollToPosition(0)
        }
        super.recreate()
    }
}
